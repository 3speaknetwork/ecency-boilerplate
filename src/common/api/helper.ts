import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
