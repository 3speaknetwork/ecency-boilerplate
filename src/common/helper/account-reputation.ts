export default (input: string | number): number => {

  if(typeof input === 'string'){
    input = Number(input);
  }
  
  if (isFloat(input)) {
    return Math.floor(input);
  }
  
  if (input === 0) {
    return 25;
  }

  if (!input) {
    return input;
  }

  let neg = false;

  if (input < 0) neg = true;

  let reputationLevel = Math.log10(Math.abs(input));
  reputationLevel = Math.max(reputationLevel - 9, 0);

  if (reputationLevel < 0) reputationLevel = 0;

  if (neg) reputationLevel *= -1;

  reputationLevel = reputationLevel * 9 + 25;

  return Math.floor(reputationLevel);
};

function isFloat(n: string | number) {
  return Number(n) === n && n % 1 !== 0;
}
