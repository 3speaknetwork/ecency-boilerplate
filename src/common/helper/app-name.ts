interface ObjInput {
    name?: string
}

export default (input: string | null | undefined | ObjInput) => {
    if (!input) {
        return '';
    }

    if (typeof input === 'string') {
        return input;
    }

    if (typeof input === 'object' && input.name !== undefined) {
        return input.name;
    }

    return '';
};
