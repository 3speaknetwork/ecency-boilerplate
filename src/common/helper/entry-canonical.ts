import {Entry} from "../store/entries/types";

import apps from "@hiveio/hivescript/apps.json";

import defaults from "../../common/constants/defaults.json";

export default (entry: Entry): string | null => {
    if (entry.json_metadata?.canonical_url) {
        return entry.json_metadata?.canonical_url;
    }

    let scheme = `${defaults.base}/{category}/@{username}/{permlink}`;

    if (entry.json_metadata?.app) {
        const identifier = entry.json_metadata.app.split("/")[0];

        if (apps[identifier]) {
            scheme = apps[identifier].url_scheme;
        }
    }

    if (!scheme) {
        return null;
    }

    return scheme
        .replace("{category}", entry.category)
        .replace("{username}", entry.author)
        .replace("{permlink}", entry.permlink);
};
