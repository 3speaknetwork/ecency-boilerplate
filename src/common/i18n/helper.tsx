import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
