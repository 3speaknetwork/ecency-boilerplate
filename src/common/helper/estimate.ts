import { votingPower } from "../api/hive";
import { ActiveUser } from "../store/active-user/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Entry } from "../store/entries/types";
import parseAsset from "./parse-asset";

export const estimate = (
  entry: Entry,
  activeUser: ActiveUser,
  dynamicProps: DynamicProps,
  percent: number
): number => {
  if (!activeUser) {
    return 0;
  }

  const { fundRecentClaims, fundRewardBalance, base, quote } = dynamicProps;
  const { data: account } = activeUser;

  if (!account.__loaded) {
    return 0;
  }

  const sign = percent < 0 ? -1 : 1;
  const postRshares = entry.net_rshares;

  const totalVests =
    parseAsset(account.vesting_shares).amount +
    parseAsset(account.received_vesting_shares).amount -
    parseAsset(account.delegated_vesting_shares).amount;

  const userVestingShares = totalVests * 1e6;

  const userVotingPower = votingPower(account) * Math.abs(percent);
  const voteEffectiveShares = userVestingShares * (userVotingPower / 10000) * 0.02;

  // reward curve algorithm (no idea whats going on here)
  const CURVE_CONSTANT = 2000000000000;
  const CURVE_CONSTANT_X4 = 4 * CURVE_CONSTANT;
  const SQUARED_CURVE_CONSTANT = CURVE_CONSTANT * CURVE_CONSTANT;

  const postRsharesNormalized = postRshares + CURVE_CONSTANT;
  const postRsharesAfterVoteNormalized = postRshares + voteEffectiveShares + CURVE_CONSTANT;
  const postRsharesCurve =
    (postRsharesNormalized * postRsharesNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + CURVE_CONSTANT_X4);
  const postRsharesCurveAfterVote =
    (postRsharesAfterVoteNormalized * postRsharesAfterVoteNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + voteEffectiveShares + CURVE_CONSTANT_X4);

  const voteClaim = postRsharesCurveAfterVote - postRsharesCurve;

  const proportion = voteClaim / fundRecentClaims;
  const fullVote = proportion * fundRewardBalance;

  const voteValue = fullVote * (base / quote);

  if (sign > 0) {
    return Math.max(voteValue * sign, 0);
  }

  return voteValue * sign;
};
