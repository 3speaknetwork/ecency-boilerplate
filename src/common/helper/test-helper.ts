import { State as GlobalState, EntryFilter, Theme, ListStyle } from "../store/global/types";
import { Entry } from "../store/entries/types";
import { Community } from "../store/community/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Vote, VestingDelegation } from "../../common/api/hive";

export const globalInstance: GlobalState = {
  filter: EntryFilter.hot,
  tag: "",
  theme: Theme.day,
  listStyle: ListStyle.row,
  intro: true,
  currency: "usd",
  currencyRate: 1,
  currencySymbol: "$",
};

export const entryInstance1: Entry = {
  post_id: 86342505,
  author: "good-karma",
  permlink: "awesome-hive",
  category: "hive",
  title: "Awesome Hive",
  body:
    "Hey developers,\n\n![awesome-hive.png](https://images.esteem.app/DQmetNmv6rtcRxueXJSNm7ErLLKNYGJETxsNSoehn6xk9BC/awesome-hive.png)\n\nI just created [Awesome-Hive](https://github.com/ledgerconnect/awesome-hive), list of services and apps on Hive.\n\nIf you don't know about Awesome project, take a look at here: https://project-awesome.org/\n\nProject contains list of services and links to various tools and libraries. Quite useful if you are developing or learning some new programming languages or skills. So we now have `awesome-hive` list, feel free to submit your pull request to add your apps, services. \n\nI will submit PR to project awesome in 30 days to list Hive in decentralized systems section. One of the requirements, awesome list should be around for more than 30 days. So I ask anyone with service, tools, dapps to submit pull request to https://github.com/ledgerconnect/awesome-hive so we have complete list of services by end of June.\n\nIn 30 days, we will PR to official project awesome list. More awareness about awesome Hive! Meanwhile, let's complete `awesome-hive` together! \ud83d\ude0e \ud83d\ude47 \ud83d\ude4f \n\n## Hive on!",
  json_metadata: {
    tags: ["hive", "hiveio", "awesome", "devs", "development", "list"],
    app: "esteem/2.2.7-surfer",
  },
  created: "2020-06-01T05:53:33",
  updated: "2020-06-01T14:41:15",
  depth: 0,
  children: 31,
  net_rshares: 234863836438976,
  is_paidout: false,
  payout_at: "2020-06-08T05:53:33",
  payout: 125.095,
  pending_payout_value: "125.095 HBD",
  author_payout_value: "0.000 HBD",
  curator_payout_value: "0.000 HBD",
  promoted: "0.000 HBD",
  replies: [],
  active_votes: [
    {
      voter: "user1",
      rshares: "34273117581576",
    },
    {
      voter: "user2",
      rshares: "2269348064337",
    },
    {
      voter: "user3",
      rshares: "19969726098",
    },
    {
      voter: "user4",
      rshares: "725359024516",
    },
    {
      voter: "user5",
      rshares: "24029616493",
    },
    {
      voter: "user6",
      rshares: "1485954337",
    },
    {
      voter: "user7",
      rshares: "1281560607198",
    },
    {
      voter: "user8",
      rshares: "554323413016",
    },
    {
      voter: "user9",
      rshares: "140063022903",
    },
    {
      voter: "user10",
      rshares: "542424211",
    },
  ],
  author_reputation: 76.58,
  stats: {
    hide: false,
    gray: false,
    total_votes: 10,
    flag_weight: 1.0,
  },
  beneficiaries: [
    {
      account: "esteemapp",
      weight: 300,
    },
  ],
  max_accepted_payout: "1000000.000 HBD",
  percent_steem_dollars: 10000,
  url: "/hive/@good-karma/awesome-hive",
  blacklists: [],
};

export const votesInstance1: Vote[] = [
  {
    voter: "user1",
    weight: 40289,
    rshares: "84880616925",
    percent: 40,
    time: "2020-06-14T15:19:51",
  },
  {
    voter: "user2",
    weight: 0,
    rshares: "39117096",
    percent: 10000,
    time: "2020-06-14T15:15:15",
  },
];

export const vestingDelegationInstance1: VestingDelegation[] = [
  {
    id: 1327252,
    delegator: "esteemapp",
    delegatee: "user1",
    vesting_shares: "14848.000000 VESTS",
    min_delegation_time: "2020-02-24T09:53:57",
  },
  {
    id: 1329459,
    delegator: "esteemapp",
    delegatee: "user2",
    vesting_shares: "191778.000000 VESTS",
    min_delegation_time: "2020-03-13T08:34:57",
  },
  {
    id: 1325910,
    delegator: "esteemapp",
    delegatee: "user3",
    vesting_shares: "29326.000000 VESTS",
    min_delegation_time: "2020-02-18T18:32:06",
  },
  {
    id: 1325312,
    delegator: "esteemapp",
    delegatee: "user4",
    vesting_shares: "27751.000000 VESTS",
    min_delegation_time: "2020-02-17T09:28:12",
  },
  {
    id: 1330486,
    delegator: "joythewanderer",
    delegatee: "user5",
    vesting_shares: "1957.888055 VESTS",
    min_delegation_time: "2020-03-24T12:42:39",
  },
];

export const discussionInstace1: Entry[] = [
  {
    post_id: 86423250,
    author: "esteemapp",
    permlink: "esteem-discord-monthly-giveaway-winners-21",
    category: "esteem",
    title: "Esteem Discord Monthly Giveaway Winners #21",
    body: "...",
    json_metadata: {},
    created: "2020-06-06T05:37:51",
    updated: "2020-06-06T05:37:51",
    depth: 0,
    children: 9,
    net_rshares: 23953037130835,
    is_paidout: false,
    payout_at: "2020-06-13T05:37:51",
    payout: 11.432,
    pending_payout_value: "11.432 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 75.6,
    stats: { hide: false, gray: false, total_votes: 180, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21",
    blacklists: [],
  },
  {
    post_id: 86437193,
    author: "forykw",
    permlink: "re-esteemapp-202067t12246786z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Prizes have consistently been very effective from @esteemapp nicely done!",
    json_metadata: {},
    created: "2020-06-07T00:02:48",
    updated: "2020-06-07T00:02:48",
    depth: 1,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-14T00:02:48",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 64.87,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@forykw/re-esteemapp-202067t12246786z",
    blacklists: [],
  },
  {
    post_id: 86427694,
    author: "brittandjosie",
    permlink: "re-esteemapp-qbi6t3",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "@ardpien congrats on being one of the winners ",
    json_metadata: { tags: ["esteem"], app: "peakd/2020.05.5" },
    created: "2020-06-06T12:03:03",
    updated: "2020-06-06T12:03:03",
    depth: 1,
    children: 1,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-13T12:03:03",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: ["ardpien/re-brittandjosie-qbjoep"],
    active_votes: [],
    author_reputation: 71.78,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@brittandjosie/re-esteemapp-qbi6t3",
    blacklists: [],
  },
  {
    post_id: 86426904,
    author: "ardpien",
    permlink: "qbi303",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body:
      "ðŸ˜ƒ I am mentioned here. That's great! All thanks to ESTEEM for creating an awesome Discord server for us to talk in there and participate in activities. Thank You ESTEEM team :) and @good-karma.",
    json_metadata: {},
    created: "2020-06-06T10:40:54",
    updated: "2020-06-06T10:40:54",
    depth: 1,
    children: 0,
    net_rshares: 114901996313,
    is_paidout: false,
    payout_at: "2020-06-13T10:40:54",
    payout: 0.031,
    pending_payout_value: "0.031 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "114901996313" }],
    author_reputation: 47.13,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@ardpien/qbi303",
    blacklists: [],
  },
  {
    post_id: 86426429,
    author: "iliyan90",
    permlink: "re-esteemapp-202066t124616772z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "@esteemapp\nBug report [click here](https://esteem.app/esteem/@iliyan90/bug-report-to-esteemapp)",
    json_metadata: {},
    created: "2020-06-06T09:46:15",
    updated: "2020-06-06T09:46:36",
    depth: 1,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-13T09:46:15",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 60.22,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@iliyan90/re-esteemapp-202066t124616772z",
    blacklists: [],
  },
  {
    post_id: 86425273,
    author: "trincowski",
    permlink: "re-esteemapp-202066t84137915z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Awesome. Thank you very much.",
    json_metadata: {},
    created: "2020-06-06T07:41:39",
    updated: "2020-06-06T07:41:39",
    depth: 1,
    children: 0,
    net_rshares: 116060350556,
    is_paidout: false,
    payout_at: "2020-06-13T07:41:39",
    payout: 0.032,
    pending_payout_value: "0.032 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "116060350556" }],
    author_reputation: 68.52,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@trincowski/re-esteemapp-202066t84137915z",
    blacklists: [],
  },
  {
    post_id: 86423472,
    author: "foxkoit",
    permlink: "re-esteemapp-202066t8590757z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Thank you ðŸ˜€ðŸ˜‡ðŸ˜‡",
    json_metadata: {},
    created: "2020-06-06T05:59:03",
    updated: "2020-06-06T05:59:03",
    depth: 1,
    children: 0,
    net_rshares: 117230405346,
    is_paidout: false,
    payout_at: "2020-06-13T05:59:03",
    payout: 0.032,
    pending_payout_value: "0.032 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [{ voter: "jznsamuel", rshares: "117230405346" }],
    author_reputation: 71.51,
    stats: { hide: false, gray: false, total_votes: 1, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@foxkoit/re-esteemapp-202066t8590757z",
    blacklists: [],
  },
  {
    post_id: 86423446,
    author: "irisworld",
    permlink: "re-esteemapp-202066t14588243z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "yahoo! thanks a lot!",
    json_metadata: {},
    created: "2020-06-06T05:58:09",
    updated: "2020-06-06T05:58:09",
    depth: 1,
    children: 0,
    net_rshares: 265974487101,
    is_paidout: false,
    payout_at: "2020-06-13T05:58:09",
    payout: 0.075,
    pending_payout_value: "0.075 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 65.01,
    stats: { hide: false, gray: false, total_votes: 2, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@irisworld/re-esteemapp-202066t14588243z",
    blacklists: [],
  },
  {
    post_id: 86423439,
    author: "behiver",
    permlink: "re-esteemapp-202066t85739415z",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body:
      "Great to see my name on the Giveaway and I like quite much engaging on Discord and find out what others are doing in the HIVE space. A source of news, common hobbies and other topics that rise up your day.",
    json_metadata: {},
    created: "2020-06-06T05:57:39",
    updated: "2020-06-06T05:57:39",
    depth: 1,
    children: 0,
    net_rshares: 273072542004,
    is_paidout: false,
    payout_at: "2020-06-13T05:57:39",
    payout: 0.077,
    pending_payout_value: "0.077 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [
      { voter: "foxkoit", rshares: "154659337454" },
      { voter: "jznsamuel", rshares: "118413204550" },
    ],
    author_reputation: 57.66,
    stats: { hide: false, gray: false, total_votes: 2, flag_weight: 0 },
    beneficiaries: [{ account: "esteemapp", weight: 300 }],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "esteemapp",
    parent_permlink: "esteem-discord-monthly-giveaway-winners-21",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@behiver/re-esteemapp-202066t85739415z",
    blacklists: [],
  },
  {
    post_id: 86440916,
    author: "ardpien",
    permlink: "re-brittandjosie-qbjoep",
    category: "esteem",
    title: "RE: Esteem Discord Monthly Giveaway Winners #21",
    body: "Thank you, Miss. @birttandjosie.",
    json_metadata: { tags: ["esteem"], app: "peakd/2020.05.5" },
    created: "2020-06-07T07:20:51",
    updated: "2020-06-07T07:20:51",
    depth: 2,
    children: 0,
    net_rshares: 0,
    is_paidout: false,
    payout_at: "2020-06-14T07:20:51",
    payout: 0,
    pending_payout_value: "0.000 HBD",
    author_payout_value: "0.000 HBD",
    curator_payout_value: "0.000 HBD",
    promoted: "0.000 HBD",
    replies: [],
    active_votes: [],
    author_reputation: 47.13,
    stats: { hide: false, gray: false, total_votes: 0, flag_weight: 0 },
    beneficiaries: [],
    max_accepted_payout: "1000000.000 HBD",
    percent_steem_dollars: 10000,
    parent_author: "brittandjosie",
    parent_permlink: "re-esteemapp-qbi6t3",
    url: "/esteem/@esteemapp/esteem-discord-monthly-giveaway-winners-21#@ardpien/re-brittandjosie-qbjoep",
    blacklists: [],
  },
];

export const communityInstance1: Community = {
  id: 1369030,
  name: "hive-148441",
  title: "GEMS",
  about: "To highlight true Gems of Hive community with User Retention as primary objective.",
  lang: "en",
  type_id: 1,
  is_nsfw: false,
  subscribers: 4086,
  sum_pending: 17370,
  num_pending: 12481,
  num_authors: 1796,
  created_at: "2020-02-26 11:33:33",
  avatar_url: "",
  description:
    "This is a collective curation community formed by @appreciator, @rocky1 and @upmewhale. You can post your genuine and creative content in this community to get more support.\n\n\nYou can post in languages other than English.\n\nThere are not many restrictions but just few simple ones and we expect you to adhere to them\n\nJoin our discord group to keep yourself updated:\nhttps://discord.gg/n98Kpmm",
  flag_text: "Plagiarism will be highly discouraged.\nHate Speech is not allowed",
  settings: {},
  team: [
    ["hive-148441", "owner", ""],
    ["bluemist", "admin", ""],
  ],
};

export const dynamicPropsIntance1: DynamicProps = {
  hivePerMVests: 513.7574961580294,
  base: 0.252,
  quote: 1,
  fundRewardBalance: 848692.443,
};
