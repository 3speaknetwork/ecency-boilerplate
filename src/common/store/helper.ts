import {Store} from "redux";

const getSymbolFromCurrency = require("currency-symbol-map");

import {AppState} from "./index";
import {ActiveUser, UserPoints} from "./active-user/types";
import {loginAct as loginActiveUser, updateAct as updateActiveUserAct} from "./active-user";

import {getAccount, getDynamicProps} from "../api/hive";
import {getPoints, usrActivity, getPromotedEntries} from "../api/private";
import {reloadAct as reloadUsers} from "./users";
import {reloadAct as reloadReblogs} from "./reblogs";
import {fetchedAct as loadDynamicProps} from "./dynamic-props";
import {fetchedAct as entriesFetchedAct} from "../../common/store/entries";
import {setCurrencyAct as setCurrency, muteNotificationsAct as muteNotifications} from "./global";

import {getCurrencyRate} from "../api/misc";

import currencies from "../constants/currencies.json";

import * as ls from "../../common/util/local-storage";

export const activeUserMaker = (name: string, points: string = "0.000", uPoints: string = "0.000"): ActiveUser => {
    return {
        username: name,
        data: {name},
        points: {
            points,
            uPoints
        }
    }
}

export const activeUserUpdater = async (store: Store<AppState>) => {
    const state = store.getState();
    if (state.activeUser) {
        const {username} = state.activeUser;

        let account;
        try {
            account = await getAccount(username);
        } catch (e) {
            return;
        }

        let points: UserPoints;
        try {
            const p = await getPoints(username);
            points = {points: p.points, uPoints: p.unclaimed_points};
        } catch (e) {
            points = {
                points: "0.000",
                uPoints: "0.000"
            };
        }

        store.dispatch(updateActiveUserAct(account, points));
    }
};

export const clientStoreTasks = (store: Store<AppState>) => {
    // Initial state from browser's local storage
    store.dispatch(reloadUsers());
    store.dispatch(loginActiveUser());
    store.dispatch(reloadReblogs());

    // Load dynamic props
    getDynamicProps().then((resp) => {
        store.dispatch(loadDynamicProps(resp));
    });

    // Update active user in interval
    activeUserUpdater(store).then();
    setInterval(() => {
        activeUserUpdater(store).then();
    }, 60 * 1000);

    // Do check-in in interval
    const checkIn = () => {
        const state = store.getState();
        if (state.activeUser) {
            usrActivity(state.activeUser?.username!, 10).then();
        }
    }
    checkIn();
    setInterval(checkIn, 1000 * 60 * 15 + 8);

    // Inject / update promoted entries to store
    const promotedEntries = () => {
        getPromotedEntries().then(r => {
            store.dispatch(entriesFetchedAct("__promoted__", r, false));
        })
    }
    promotedEntries();
    setInterval(promotedEntries, 1000 * 60 * 5);

    // Currency
    const currency = ls.get("currency");
    if (currency && currencies.find(x => x.id === currency)) {
        const symbol = getSymbolFromCurrency(currency);
        getCurrencyRate(currency).then(rate => {
            store.dispatch(setCurrency(currency, rate, symbol));
        });
    }

    // Notifications
    if (ls.get("notifications") === false) {
        store.dispatch(muteNotifications());
    }
}
