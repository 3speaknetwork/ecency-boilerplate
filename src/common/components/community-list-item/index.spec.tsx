import React from "react";

import CommunityListItem from "./index";
import TestRenderer from "react-test-renderer";

import { createBrowserHistory } from "history";
import { StaticRouter } from "react-router-dom";

import { communityInstance1 } from "../../helper/test-helper";

it("(1) Default render", () => {
  const props = {
    history: createBrowserHistory(),
    community: { ...communityInstance1 },
    addAccount: () => {},
  };
  
  const comp = <CommunityListItem {...props} />;

  const renderer = TestRenderer.create(
    <StaticRouter location="/@username" context={{}}>
      {comp}
    </StaticRouter>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});
