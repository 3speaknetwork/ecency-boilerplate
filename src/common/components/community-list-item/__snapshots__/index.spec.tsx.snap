// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="community-list-item"
>
  <div
    className="item-content"
  >
    <h2
      className="item-title"
    >
      <span
        className="user-avatar small"
        style={
          Object {
            "backgroundImage": "url(https://images.ecency.com/u/hive-148441/avatar/small)",
          }
        }
      />
      <a
        href="/trending/hive-148441"
        onClick={[Function]}
      >
        GEMS
      </a>
    </h2>
    <div
      className="item-about"
    >
      To highlight true Gems of Hive community with User Retention as primary objective.
    </div>
    <div
      className="item-stats"
    >
      <div
        className="stat"
      >
        4,086 subscribers
      </div>
      <div
        className="stat"
      >
        1,796 posters
      </div>
      <div
        className="stat"
      >
        12,481 posts
      </div>
    </div>
  </div>
  <div
    className="item-controls"
  >
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Subscribe
    </button>
  </div>
</div>
`;
